protocol/testutil/app/app.go

```diff
@@ -382,49 +455,181 @@ func (tApp *TestApp) initChainIfNeeded() {
 	originalServerStartTime := srvtypes.ServerStartTime.Load()
 	srvtypes.ServerStartTime.Store(int64(time.Millisecond * 10))
 
-	validatorHomeDir, app, shutdownFn, err := launchValidator(tApp.genesis, tApp.builder.appOptions)
-	if err != nil {
-		tApp.builder.t.Fatal(err)
-	}
-	tApp.App = app
+	// Launch the main instance of the application
+	{
```

Comment:
The issue that I didn't find a good way to handle the `restart fn` since I rely on updating the references that are captured outside which are shared between the `restart fn` and `cleanup fn`.

I filed https://linear.app/dydx/issue/CORE-721